resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/VolcanoFeatureStep.scala

```diff
@@ -59,6 +55,7 @@ private[spark] class VolcanoFeatureStep extends KubernetesDriverCustomFeatureCon
 
     var spec = pg.getSpec
     if (spec == null) spec = new PodGroupSpec
+    // Overwrite by driver `pod.spec.priorityClassName`
```

Comment:
Hmm. Do you use two set of priorityClass? In other words, does your system have the following separately?
- priority classes for Pod
- priority classes for PodGroup (volcano)

Reply:
For type of priority class, it's the same thing, pod/pg are share the [priorityClass](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) in Kubernetes.

Most of users are using same priorities series with pod to make sure job and pod scheduling can be considered in a whole.

But I also heard some users want to set a separate priority classes for PodGroup. This is also why I raised my concern before on https://github.com/apache/spark/pull/35776#discussion_r822247430 .

Reply:
Ya, that's the reason why I asked you again about that.
1. Do you know the background of separate priority classes for PodGroup?
2. If all custom schedulers really want to use their own priorityClasses, we need to adjust our implementation a little again.

Reply:
> Do you know the background of separate priority classes for PodGroup?

- A case I heard before is [multiple sheduler](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-multiple-schedulers/), for some reason they want to separate priority class series for differet scheduler. So they want to a separate priority for each scheudlers.
- The other case is pod level priority in same podgroup, they can set different priority for pod. But I think it is not for spark single driver case.

These cases are not very common, but I think the biggest concern in here is simpfied configuration, such as if users want to use volcano priority scheduling in a queue, they have to specify two configuration: the pod template to set priority and the podgroup template to set queue. Generally speaking, they prefer to accept only maintain a podgroup template for resource level scheduling configuration.

So overwrite default priority if user specify priority in podgroup template might be a alternate choice.

Reply:
Ya, that sounds fair. I also support more isolations, @Yikun .

Reply:
Here is the PR, https://github.com/apache/spark/pull/35807 .